FirebaseCore/Sources/FIRApp.m

```diff
@@ -883,4 +886,33 @@ - (void)appDidBecomeActive:(NSNotification *)notification {
   }
 }
 
+#if DEBUG
++ (void)findMisnamedGoogleServiceInfoPlist {
+  for (NSBundle *bundle in [NSBundle allBundles]) {
+    // Not recursive, but we're looking for misnames, not people accidentally
+    // hiding their config file in a subdirectory of their bundle.
+    // This is only really designed for iOS.
```

Comment:
Any reason you suspect this won't work on other platforms? I _think_ `pathsForResourcesOfType:inDirectory:` searches the `Resources` subdirectory on macOS (where the `GoogleService-Info.plist` file lives) when `inDirectory` is `nil` based on the [discussion](https://developer.apple.com/documentation/foundation/nsbundle/1413058-pathsforresourcesoftype#discussion) in the docs.

Reply:
It should work for all platforms, but macOS bundles may make use of subdirectories more frequently. I can remove this line if it's misleading.